tests/unit_tests/dft/include/reference_dft.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef ONEMKL_REFERENCE_DFT_HPP
+#define ONEMKL_REFERENCE_DFT_HPP
+
+#include <sycl/sycl.hpp>
+
+using namespace oneapi::mkl;
+using namespace sycl;
+
+template <typename T>
+struct is_complex_t : public std::false_type {};
+
+template <typename T>
+struct is_complex_t<std::complex<T>> : public std::true_type {};
+
+template <typename TypeIn, typename TypeOut>
+bool reference_forward_dft(sycl::queue &queue, sycl::buffer<TypeIn, 1> &in,
```

Comment:
At first I was using the GPU to calculate the reference. My gpu cannot do the calculations using doubles which makes the results very imprecise. Moving the calculations to the host and using double improved it substantially but it's still quite a bit off. 

Not sure how to improve it further. It might be just imprecisions that result from having very different implementation (fft vs naive implementation)

Reply:
I think it's fine to keep the reference on the host. In that case you can remove the copies between host and device before and inside `reference_forward_dft`.
Regarding the precision issue, the error margin of 200 is not necessarily an issue. Some of the blas test use a high margin like 1357 for `dot` or `10 * k` for GEMM. That margin is multiplied by the numeric_limit epsilon so it is quite a small margin.

Reply:
Those are good points. 200 is probably good enough. Before (with the GPU version that does not calculate using double) I had to use something like 10000. That I thought was too much.